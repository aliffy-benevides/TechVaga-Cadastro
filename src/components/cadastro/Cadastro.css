.cadastro {
    background-color: #fff;
    border-radius: 16px;
    box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
    padding: 20px;
}

.cadastro form {
    text-align: center;
}

.cadastro form .form-group label {
    font-size: 1.2em;
}