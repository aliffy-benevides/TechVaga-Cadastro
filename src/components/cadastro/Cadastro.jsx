import React from 'react'
import './Cadastro.css'

const { REACT_APP_API_BASE_URL: apiBaseUrl } = process.env

export default props => {
    return (
        <div className="cadastro">
            <form name="cadastro">
                <div className="form-group">
                    <label htmlFor="vagas_atualizadas">Quantidade de Vagas</label>
                    <input type="number" name="vacancies" className="form-control" id="vagas_atualizadas" min={0} />
                </div>
                {/* <input name="vacancies" id="vagas_atualizadas" /> */}
                <button id="atualizar_vagas" type="submit" className="btn btn-info">
                    Atualizar
                </button>
            </form>
        </div>
    )
}
