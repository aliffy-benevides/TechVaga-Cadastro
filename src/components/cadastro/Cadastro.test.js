import React from "react"
import { render, unmountComponentAtNode } from "react-dom"
import { act } from "react-dom/test-utils"
import axios from 'axios'

import Cadastro from './Cadastro'

const { REACT_APP_API_BASE_URL: apiBaseUrl } = process.env
jest.mock('axios')

let container = null
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div")
    document.body.appendChild(container)
})

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container)
    container.remove()
    container = null
})

it("update total vacancies", async () => {
    const url = `${apiBaseUrl}/vacancies`
    const totalVacancies = 200
    axios.post.mockImplementationOnce(() => {
        return Promise.resolve("OK")
    })
    
    await act(async () => {
        render(<Cadastro />, container)
    })

    const input = container.querySelector("[name='vacancies']")
    const button = container.querySelector("#atualizar_vagas")
    expect(input).toMatchSnapshot()
    expect(button).toMatchSnapshot()
})
