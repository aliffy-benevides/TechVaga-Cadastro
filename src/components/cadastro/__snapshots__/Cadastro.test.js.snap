// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update total vacancies 1`] = `
<input
  class="form-control"
  id="vagas_atualizadas"
  min="0"
  name="vacancies"
  type="number"
/>
`;

exports[`update total vacancies 2`] = `
<button
  class="btn btn-info"
  id="atualizar_vagas"
  type="submit"
>
  Atualizar
</button>
`;
